File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const hashMap = new Map()
        for (let i=0; i <= nums.length; i++) {
            const complementIdx = hashMap.get(nums[i]) 
            if (complementIdx !== undefined) return [complementIdx, i]

            hashMap.set(target - nums[i], i)
        }
        return []
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int, offset = 0) -> int:
        if not nums: return -1
        pointer = int(len(nums) / 2)
        if nums[pointer] == target:
            return pointer + offset
        if nums[pointer] < target:
            return self.search(nums[pointer+1:], target, offset+pointer+1)
        else:
            return self.search(nums[:pointer], target, offset)